src/ralph/backends/lrs/async_es.py

```diff
@@ -0,0 +1,110 @@
+"""Elasticsearch LRS backend for Ralph."""
+
+import logging
+from typing import Iterator, List
+
+from ralph.backends.data.async_es import AsyncESDataBackend, ESQuery, ESQueryPit
+from ralph.backends.lrs.base import (
+    AgentParameters,
+    BaseAsyncLRSBackend,
+    StatementParameters,
+    StatementQueryResult,
+)
+from ralph.exceptions import BackendException, BackendParameterException
+
+logger = logging.getLogger(__name__)
+
+
+class AsyncESLRSBackend(BaseAsyncLRSBackend, AsyncESDataBackend):
+    """Asynchronous Elasticsearch LRS backend implementation."""
+
+    settings_class = AsyncESDataBackend.settings_class
+
+    async def query_statements(
+        self, params: StatementParameters
+    ) -> StatementQueryResult:
+        """Return the statements query payload using xAPI parameters."""
+        es_query_filters = []
+
+        if params.statementId:
+            es_query_filters += [{"term": {"_id": params.statementId}}]
+
+        self._add_agent_filters(es_query_filters, params.agent, "actor")
+        self._add_agent_filters(es_query_filters, params.authority, "authority")
+
+        if params.verb:
+            es_query_filters += [{"term": {"verb.id.keyword": params.verb}}]
+
+        if params.activity:
+            es_query_filters += [
+                {"term": {"object.objectType.keyword": "Activity"}},
+                {"term": {"object.id.keyword": params.activity}},
+            ]
+
+        if params.since:
+            es_query_filters += [{"range": {"timestamp": {"gt": params.since}}}]
+
+        if params.until:
+            es_query_filters += [{"range": {"timestamp": {"lte": params.until}}}]
+
+        es_query = {
+            "pit": ESQueryPit.construct(id=params.pit_id),
+            "size": params.limit,
+            "sort": [{"timestamp": {"order": "asc" if params.ascending else "desc"}}],
+        }
+        if len(es_query_filters) > 0:
+            es_query["query"] = {"bool": {"filter": es_query_filters}}
+
+        if params.ignore_order:
+            es_query["sort"] = "_shard_doc"
+
+        if params.search_after:
+            es_query["search_after"] = params.search_after.split("|")
+
+        # Note: `params` fields are validated thus we skip their validation in ESQuery.
+        query = ESQuery.construct(**es_query)
+        try:
+            es_documents = [
+                document
+                async for document in self.read(query=query, chunk_size=params.limit)
+            ]
```

Comment:
```suggestion
            statements = [
                document["_source"]
                async for document in self.read(query=query, chunk_size=params.limit)
            ]
```
Could we avoid iterating over the documents twice?